vernac/classes.ml

```diff
@@ -267,9 +264,11 @@ let new_instance ?(abstract=false) ?(global=false) ?(refine= !refine_instance)
               | (n, _) :: _ ->
 		unbound_method env' k.cl_impl (get_id n)
               | _ ->
-		Some (Inl (type_ctx_instance evars (push_rel_context ctx' env') 
-			     k.cl_props props subst))
-      in	  
+                let kcl_props = List.map (Termops.map_rel_decl of_constr) k.cl_props in
+                let _evd = ref sigma in
+                Some (Inl (type_ctx_instance (push_rel_context ctx' env') _evd
+                             kcl_props props subst)), !_evd
```

Comment:
This line is broken in flambda, it returns non-modified `sigma` when we want the one modified by runnning `type_ctx_instance` (at least in ocamldebug).

Reply:
Confirmed that let-lifting the function application fixes the missing constraint in crelationclasses.

Reply:
This is typically a case where both evaluation orders are valid, so you should left-lift the `type_ctx` application indeed if it needs to be performed before `!_evd` is computed.

Reply:
Oh indeed that was my fault; I was not so careful as I was planning to remove that code anyways.

Reply:
In fact the `ref` and the `_evd` are already gone in the next PR.